scripts/build-shaded-dtest-jar-local.sh

```diff
@@ -26,7 +26,7 @@ CASSANDRA_VERSION=$(cat build.xml | grep 'property name="base.version"' | awk -F
 GIT_HASH=$(git rev-parse --short HEAD)
 DTEST_ARTIFACT_ID=${ARTIFACT_NAME}-local
 DTEST_JAR_DIR="$(dirname "${SCRIPT_DIR}/")/dtest-jars"
-
+DTEST_JAR_DIR=${CASSANDRA_DEP_DIR:-$DTEST_JAR_DIR}
```

Comment:
I do not think the dtest jars should be placed in a central place and shared. The projects can depend on different SHAs. Unlike the local maven repo, where the jars can be distinguished by version. Sharing the dtest jars could cause confusion. 
It is unrelated to the patch, just to bring it up, do not need to address.

Reply:
This builds a jar with `dtest-M.m.p.jar` in the shared directory _if you set the environment variable_, and the local dir if not - either way, they files are versioned. If you happen to use the shared dependencies directory, you often don't have to rebuild C* versions. A previous patch started to _use_ the shared directory in _gradle_ but didn't include it here, so I made the two consistent. I happen to like not having several dozen copies of C* builds floating around my system personally, so I use it and have never had issues... just don't set the `CASSANDRA_DEP_DIR` environment variable if you don't want to use it.